jax/experimental/sparse/linalg.py

```diff
@@ -291,20 +294,51 @@ def _eigh_ascending(A):
 
 
 def _svqb(X):
-  # https://sdm.lbl.gov/~kewu/ps/45577.html
-  # May choose to truncate the basis rather than return
-  # a degenerate, non-orthogonal one.
+  """Derives a truncated orthonormal basis for `X`.
 
+  SVQB [1] is an accelerator-friendly orthonormalization procedure, which
+  squares the matrix `C = X.T @ X` and computes an eigenbasis for a smaller
```

Comment:
Rank revealing QR would be a nice options if it was available in JAX. It might be worth adding. A version based on the randomized projection approach could be made accelerator-friendly.

Reply:
By QR here, you mean QR-ing `X` itself not its square right? At one point early in the algorithm development process I tried that but decided against it b/c the speed wasn't satisfactory. Cholesky on XTX would be faster but rank deficiency is specifically a case that needs to be handled here.

That said, I'll put trying the QR approach out again on my TODO list. It's worth revisiting and I don't have hard numbers saying it's too slow. And it avoids squaring.

I'm really intrigued by the randomized projection you're mentioning. Do you have a reference?

Reply:
Here is a reference for the RRQR with randomized projection: https://arxiv.org/abs/2008.04447

You are right that computing the QR decomposition of X is likely slower, since `X.T@X` is so fast on an accelerator. But avoiding the squaring would be nice. Just some things to experiment with, I guess.